### DISCUSSION TITLE
Custom Search Contexts

### DISCUSSION THREAD

**Discussion by axieax** [axieax/urlview.nvim#40] [Show and tell]
Feel free to share your custom search contexts here (may become a built-in as well)!

**Comment by axieax** [comment]
Custom Jira Tickets (from the docs):

```lua
local search = require("urlview.search")
local search_helpers = require("urlview.search.helpers")
search["jira"] = search_helpers.generate_custom_search({
  capture = "AXIE%-%d+",
  format = "https://jira.axieax.com/browse/%s",
})
```

This captures Jira tickets such as "AXIE-1", "AXIE-17", "AXIE-132".

**Comment by lleger** [comment]
Search for Hex packages from `mix.exs` (Elixir ecosystem package manager):

```lua
search["hex"] = search_helpers.generate_custom_search({
  capture = "{:([%a_]+),",
  format = "https://hex.pm/packages/%s",
})
```

This captures eg `{phoenix, "~> 1.5.9"}` and shows `https://hex.pm/packages/phoenix`.
